src/compiler.ts

```diff
@@ -6512,14 +6598,64 @@ export class Compiler extends DiagnosticEmitter {
         return module.unreachable();
       }
     }
-    return this.compileCallIndirect(
-      assert(signature), // FIXME: asc can't see this yet
-      indexArg,
-      expression.arguments,
-      expression,
-      0,
-      contextualType == Type.void
-    );
+
+    var returnType = signature.returnType;
+    var tempResultLocal = this.currentFlow.getTempLocal(returnType);
+    var tempFunctionReferenceLocal = this.currentFlow.getTempLocal(returnType);
```

Comment:
Added locals here to accomplish two things:
- tempFunctionReferenceLocal is used to store the result of the index argument expression- we need to make sure this is only executed once, so we store it to a local before using it multiple times in the following lines
- tempResultLocal: after we finish the `call_indirect` invokation, we need to change the global closure value back to what it was before the call began, and then we need to return whatever the call returned- since we need to access something that's below the top of the stack, afaik there is no way to do that without a local

I think there's probably a way to do this with just one local, but wanted to get my latest changes up on the PR before finishing up for the day- I'll probably look at this again tomorrow.